File: 456-132-pattern/456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) 
    {
        stack<int>s;
        
        int c=INT_MIN;
        
        for(int i=nums.size()-1;i>=0;i--)
        {
            if(nums[i]<c)
            {
                return true;
            }
            
            while(s.size()!=0 && s.top()<nums[i])
            {
                c=s.top();
                s.pop();
            }
            s.push(nums[i]);
        }
        
        return false;
    }
};

File: 456-132-pattern/456-132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        stack = deque()
        n = len(nums)
        k = -inf
        for j in range(n-1,-1,-1):
            if nums[j]<k:
                return True
            
            while stack and stack[-1]<nums[j]:
                k = stack.pop()
                
            stack.append(nums[j])
        
        return False